java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProject.java

```diff
@@ -224,19 +224,30 @@ public void setSources(List<ProjectSource> sourcesIn) {
     /**
      * Adds a source to content project
      *
-     * @param source the source
+     * @param source the Source
+     * @throws java.lang.IllegalArgumentException if the given Source does not match to the Project
+     * @return true if Source was added, false if the Source had been present already
      */
-    public void addSource(ProjectSource source) {
-        source.setContentProject(this);
-        sources.add(source);
+    public boolean addSource(ProjectSource source) {
+        if (!source.getContentProject().equals(this)) {
+            throw new IllegalArgumentException("Trying to attach source " + source + " with mismatched Project.");
```

Comment:
I am maybe too blind to figured this out but, what is the code-path/work-flow that lets this happen?

Reply:
You're right. This is unnecessary. I added the for some reason when I implemented the positioning.